api/app/Models/User.php

```diff
@@ -741,10 +745,42 @@ public function getEnrichedNotificationsAttribute()
     // rename accessor to avoid hiding parent's notification function
     public function getUnreadEnrichedNotificationsAttribute()
     {
-        $notifications = $this->unreadNotifications()->get();
+        $user = Auth::user();
+        $notifications = $this->unreadNotifications()
+            ->where('notifiable_id', $user->id)
+            ->get();
         $notifications->each(function ($n) {
             self::enrichNotification($n);
         });
         return $notifications;
     }
+
+
+    public function scopeAuthorizedToView(Builder $query)
+    {
+        $user = Auth::user();
+
+        if (!$user) {
+            return $query->where('id', null);
```

Comment:
This is a short circuit, right?  I wonder if there's a better way that doesn't issue a db request at all?  :shrug:

Reply:
I can't think of a better way to do it either 👍

Reply:
We should consider back-porting this to PoolCandidate.scopeAuthorizedToView, which will just crash if the user is null